src/pages/Admin/PermissionHandler.tsx

```diff
@@ -0,0 +1,33 @@
+import { ReactNode, useEffect, useState } from "react";
+import { matchPath, useLocation } from "react-router-dom";
+import { AdminResources } from "services/types";
+import { useGetWallet } from "contexts/WalletContext";
+import { adminRoutes, appRoutes } from "constants/routes";
+import { useAdminResources } from "./Guard";
+
+const ADMIN_ROUTE = `${appRoutes.admin}/:id/`;
+
+export default function PermissionHandler({
+  data,
+  children,
+}: {
+  data: AdminResources | undefined;
+  children: ReactNode;
+}) {
+  const { wallet } = useGetWallet();
+  const currPath = useLocation().pathname;
+  const [disableInputs, setDisableInputs] = useState(true);
+  const { whitelistedBeneficiaries, members } = useAdminResources<"charity">();
```

Comment:
```suggestion
  const { whitelistedBeneficiaries, members } = useAdminResources();
```

`PermissionHandler` is top level component so using 
`useAdminResources<"charity">` would error when `user` goes to `admin/0` (ap-team) or `admin/0.5` (review-team) since `whiteListedBeneficiaries` is not present